onnx/defs/math/defs.cc

```diff
@@ -8,62 +8,31 @@ using namespace ONNX_NAMESPACE;
 
 namespace ONNX_NAMESPACE {
 
-const char* kBroadcastDoc = R"DOC(
-If necessary the right-hand-side argument will be broadcasted to match the
-shape of left-hand-side argument. When broadcasting is specified, the second
-tensor can either be of element size 1 (including a scalar tensor and any
-tensor with rank equal to or smaller than the first tensor), or having its
-shape as a contiguous subset of the first tensor's shape. The starting of the
-mutually equal shape is specified by the argument "axis", and if it is not set,
-suffix matching is assumed. 1-dim expansion doesn't work yet.
-
-For example, the following tensor shapes are supported (with broadcast=1):
-
-  shape(A) = (2, 3, 4, 5), shape(B) = (,), i.e. B is a scalar tensor
-  shape(A) = (2, 3, 4, 5), shape(B) = (1, 1), i.e. B is an 1-element tensor
-  shape(A) = (2, 3, 4, 5), shape(B) = (5,)
-  shape(A) = (2, 3, 4, 5), shape(B) = (4, 5)
-  shape(A) = (2, 3, 4, 5), shape(B) = (3, 4), with axis=1
-  shape(A) = (2, 3, 4, 5), shape(B) = (2), with axis=0
-
-Attribute `broadcast=1` needs to be passed to enable broadcasting.
-)DOC";
-
 std::function<void(OpSchema&)> MathDocGenerator(const char* name) {
   return [=](OpSchema& schema) {
     std::string doc = R"DOC(
-Performs element-wise binary {name} (with limited broadcast support).
-{broadcast_doc})DOC";
+Performs element-wise binary {name} (with Numpy-style broadcasting support).
+
+{broadcast_doc}
+)DOC";
     ReplaceAll(doc, "{name}", name);
-    ReplaceAll(doc, "{broadcast_doc}", kBroadcastDoc);
+    ReplaceAll(doc, "{broadcast_doc}", GenerateBroadcastingDocMul().c_str());
     schema.SetDoc(doc);
-    schema.Attr(
-        "broadcast",
-        "Pass 1 to enable broadcasting",
-        AttributeProto::INT,
-        static_cast<int64_t>(0));
-    schema.Attr(
-        "axis",
-        "If set, defines the broadcast dimensions. See doc for details.",
-        AttributeProto::INT,
-        OPTIONAL);
     schema.Input(
         0,
         "A",
-        "First operand, should share the type with the second operand.",
+        "First operand.",
         "T");
     schema.Input(
         1,
         "B",
-        "Second operand. With broadcasting can be of smaller size than A. "
-        "If broadcasting is disabled it should be of the same size.",
+        "Second operand.",
         "T");
-    schema.Output(0, "C", "Result, has same dimensions and type as A", "T");
+    schema.Output(0, "C", "Result, has same element type as two inputs", "T");
     schema.TypeConstraint(
         "T",
         OpSchema::high_precision_numeric_types(),
         "Constrain input and output types to high-precision numeric tensors.");
-    schema.TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput);
```

Comment:
@gramalingam @anderspapitto  please be aware of that type and shape inference function for math ops need to be updated. it's removed in this PR :).

Reply:
@gramalingam are you interested in re-implementing these to account for the new broadcasting or shall I?

Reply:
@anderspapitto if you can, that would be great!